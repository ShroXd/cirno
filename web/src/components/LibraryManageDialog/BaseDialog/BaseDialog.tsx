import { FC, useCallback, useEffect } from 'react'
import { Controller, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'

import { FolderIcon, XMarkIcon } from '@heroicons/react/24/outline'
import {
  Button,
  Dialog,
  DialogBody,
  DialogFooter,
  DialogHeader,
  IconButton,
  Input,
  Option,
  Select,
  Typography,
} from '@material-tailwind/react'

import { LibraryDto } from '~/bindings/LibraryDto'

interface BaseDialogProps {
  title: string
  description: string
  submitButtonText: string
  open: boolean
  defaultValues?: LibraryDto

  onSubmit: (data: LibraryDto) => Promise<void>
  dialogHandler: () => void
  onClose?: () => void
}

export const BaseDialog: FC<BaseDialogProps> = ({
  title,
  description,
  submitButtonText,
  open,
  defaultValues,
  onSubmit,
  dialogHandler,
  onClose,
}) => {
  const {
    register,
    handleSubmit,
    control,
    formState: { errors },
    reset,
  } = useForm<LibraryDto>({ values: defaultValues, defaultValues })
  const { t } = useTranslation()

  const handleDialogClose = useCallback(() => {
    dialogHandler()
    onClose?.()
  }, [dialogHandler, onClose])

  useEffect(() => {
    if (open) {
      reset(defaultValues)
    }
  }, [open, reset, defaultValues])

  return (
    <Dialog
      size='sm'
      open={open}
      handler={handleDialogClose}
      dismiss={{
        outsidePress: false,
      }}
      className='p-4'
      data-testid='library-manage-dialog'
    >
      <form onSubmit={handleSubmit(onSubmit)}>
        <DialogHeader className='relative m-0 block'>
          <Typography variant='h4' color='blue-gray' data-testid='dialog-title'>
            {title}
          </Typography>
          <Typography className='mt-1 font-normal text-gray-600'>
            {description}
          </Typography>
          <IconButton
            size='sm'
            variant='text'
            className='!absolute right-3.5 top-3.5'
            onClick={handleDialogClose}
          >
            <XMarkIcon className='h-4 w-4 stroke-2' />
          </IconButton>
        </DialogHeader>

        <DialogBody className='space-y-4 pb-6'>
          <div>
            <Typography
              variant='small'
              color='blue-gray'
              className='mb-2 text-left font-medium'
            >
              {t('component.libraryManageDialog.name')}
            </Typography>
            <Input
              color='gray'
              size='lg'
              placeholder={t('component.libraryManageDialog.namePlaceholder')}
              {...register('name', { required: true })}
              className='!border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900'
              containerProps={{
                className: '!min-w-full',
              }}
              labelProps={{
                className: 'before:content-none after:content-none',
              }}
            />
            {errors.name && (
              <Typography variant='small' color='red' className='mt-1'>
                {t('component.libraryManageDialog.nameRequired')}
              </Typography>
            )}
          </div>
          <div>
            <Typography
              variant='small'
              color='blue-gray'
              className='mb-2 text-left font-medium'
            >
              {t('component.libraryManageDialog.category')}
            </Typography>
            <Controller
              name='category'
              defaultValue='Movie'
              control={control}
              rules={{ required: true }}
              render={({ field }) => (
                <Select placeholder='1' {...field}>
                  {/* TODO: get the options & its values from the rs_ts generated enum */}
                  <Option value='Movie'>
                    {t('component.libraryManageDialog.categoryOptions.movie')}
                  </Option>
                  <Option value='TvShow'>
                    {t('component.libraryManageDialog.categoryOptions.tvShow')}
                  </Option>
                  <Option value='Animation'>
                    {t(
                      'component.libraryManageDialog.categoryOptions.animation'
                    )}
                  </Option>
                </Select>
              )}
            ></Controller>
            {errors.category && (
              <Typography variant='small' color='red' className='mt-1'>
                {t('component.libraryManageDialog.categoryRequired')}
              </Typography>
            )}
          </div>
          <div className='flex flex-col gap-4'>
            <Typography
              variant='small'
              color='blue-gray'
              className='text-left font-medium'
            >
              {t('component.libraryManageDialog.directory')}
            </Typography>
            <div className='relative -mt-2 w-full'>
              <Input
                color='gray'
                size='lg'
                placeholder={t(
                  'component.libraryManageDialog.directoryPlaceholder'
                )}
                {...register('directory', { required: true })}
                className='!border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900'
                containerProps={{
                  className: '!min-w-full',
                }}
                labelProps={{
                  className: 'before:content-none after:content-none',
                }}
              />
              <div className='absolute right-1.5 top-1.5'>
                <IconButton
                  size='sm'
                  className='rounded'
                  variant='text'
                  disabled
                >
                  <FolderIcon className='h-5 w-5' />
                </IconButton>
              </div>
            </div>
            {errors.directory && (
              <Typography variant='small' color='red' className='-mt-3'>
                {t('component.libraryManageDialog.directoryRequired')}
              </Typography>
            )}
          </div>
        </DialogBody>
        <DialogFooter>
          <Button className='ml-auto' type='submit' data-testid='submit-button'>
            {submitButtonText}
          </Button>
        </DialogFooter>
      </form>
    </Dialog>
  )
}
