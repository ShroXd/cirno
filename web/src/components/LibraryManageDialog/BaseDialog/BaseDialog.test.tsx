import { ThemeProvider } from '@material-tailwind/react'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { BaseDialog } from './BaseDialog'

describe('BaseDialog', async () => {
  it('should render without a11y violations', async () => {
    const { container } = render(
      <BaseDialog
        title='Test'
        description='Test'
        submitButtonText='Test'
        open={true}
        onClose={() => {}}
        onSubmit={() => Promise.resolve()}
        dialogHandler={() => {}}
      />
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly', () => {
    const { baseElement } = render(
      <ThemeProvider>
        <BaseDialog
          title='Test title'
          description='Test description'
          submitButtonText='Test submit button text'
          open={true}
          onClose={() => {}}
          onSubmit={() => Promise.resolve()}
          dialogHandler={() => {}}
        />
      </ThemeProvider>
    )

    expect(baseElement).toMatchSnapshot()
  })
})
