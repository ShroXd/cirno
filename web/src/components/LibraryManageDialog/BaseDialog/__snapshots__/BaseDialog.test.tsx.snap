// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseDialog > should render correctly 1`] = `
<body
  data-floating-ui-scroll-lock=""
  style="overflow: hidden; padding-right: 1024px;"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-floating-ui-portal=""
    id=":r3:"
  >
    <div
      style="position: fixed; overflow: auto; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 9999;"
    >
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="grid place-items-center fixed w-screen h-screen bg-black bg-opacity-60 backdrop-blur-sm"
        style="opacity: 0.013888888888888892;"
      >
        <div
          aria-describedby=":r0:-description"
          aria-labelledby=":r0:-label"
          class="relative bg-white m-4 rounded-lg shadow-2xl text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed w-full md:w-2/3 lg:w-2/4 2xl:w-1/3 min-w-[80%] md:min-w-[66.666667%] lg:min-w-[50%] 2xl:min-w-[33.333333%] max-w-[80%] md:max-w-[66.666667%] lg:max-w-[50%] 2xl:max-w-[33.333333%] p-4"
          data-testid="library-manage-dialog"
          id=":r1:"
          role="dialog"
          style="opacity: 0.00617283950617284; transform: translateY(-49.69135802469136px) translateZ(0);"
          tabindex="-1"
        >
          <form>
            <div
              class="items-center shrink-0 p-4 text-blue-gray-900 antialiased font-sans text-2xl font-semibold leading-snug relative m-0 block"
            >
              <h4
                class="block antialiased tracking-normal font-sans text-2xl font-semibold leading-snug text-blue-gray-900"
                data-testid="dialog-title"
              >
                Test
              </h4>
              <p
                class="block antialiased font-sans text-base leading-relaxed mt-1 font-normal text-gray-600"
              >
                Test
              </p>
              <button
                class="relative align-middle select-none font-sans font-medium text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-8 max-w-[32px] h-8 max-h-[32px] rounded-lg text-xs text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20 !absolute right-3.5 top-3.5"
                type="button"
              >
                <span
                  class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 stroke-2"
                    data-slot="icon"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 18 18 6M6 6l12 12"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="relative p-4 text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed space-y-4 pb-6"
            >
              <div>
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 mb-2 text-left font-medium"
                >
                  component.libraryManageDialog.name
                </p>
                <div
                  class="relative w-full min-w-[200px] h-11 !min-w-full"
                >
                  <input
                    class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent placeholder:opacity-0 focus:placeholder:opacity-100 text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-900 !border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900"
                    name="name"
                    placeholder="component.libraryManageDialog.namePlaceholder"
                  />
                  <label
                    class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-gray-500 peer-focus:text-gray-900 before:border-blue-gray-200 peer-focus:before:!border-gray-900 after:border-blue-gray-200 peer-focus:after:!border-gray-900 before:content-none after:content-none"
                  >
                     
                  </label>
                </div>
              </div>
              <div>
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 mb-2 text-left font-medium"
                >
                  component.libraryManageDialog.category
                </p>
                <div
                  class="relative w-full min-w-[200px] h-10"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal text-left outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all border text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 border-t-transparent"
                    name="category"
                    placeholder="1"
                    role="combobox"
                    type="button"
                  >
                    <span
                      class="absolute top-2/4 -translate-y-2/4 left-3 pt-0.5"
                      value="Movie"
                    >
                      component.libraryManageDialog.categoryOptions.movie
                    </span>
                    <div
                      class="grid place-items-center absolute top-2/4 right-2 pt-px w-5 h-5 text-blue-gray-400 rotate-0 -translate-y-2/4 transition-all"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                  <label
                    class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal transition-all -top-1.5 before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 before:rounded-tl-md before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 after:rounded-tr-md after:pointer-events-none after:transition-all peer-disabled:after:border-transparent text-[11px] peer-disabled:text-transparent before:border-t before:border-l after:border-t after:border-r leading-tight text-blue-gray-400 before:border-blue-gray-200 after:border-blue-gray-200"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-4"
              >
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 text-left font-medium"
                >
                  component.libraryManageDialog.directory
                </p>
                <div
                  class="relative -mt-2 w-full"
                >
                  <div
                    class="relative w-full min-w-[200px] h-11 !min-w-full"
                  >
                    <input
                      class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent placeholder:opacity-0 focus:placeholder:opacity-100 text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-900 !border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900"
                      name="directory"
                      placeholder="component.libraryManageDialog.directoryPlaceholder"
                    />
                    <label
                      class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-gray-500 peer-focus:text-gray-900 before:border-blue-gray-200 peer-focus:before:!border-gray-900 after:border-blue-gray-200 peer-focus:after:!border-gray-900 before:content-none after:content-none"
                    >
                       
                    </label>
                  </div>
                  <div
                    class="absolute right-1.5 top-1.5"
                  >
                    <button
                      class="relative align-middle select-none font-sans font-medium text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-8 max-w-[32px] h-8 max-h-[32px] text-xs text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20 rounded"
                      disabled=""
                      type="button"
                    >
                      <span
                        class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform"
                      >
                        <svg
                          aria-hidden="true"
                          class="h-5 w-5"
                          data-slot="icon"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="1.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M2.25 12.75V12A2.25 2.25 0 0 1 4.5 9.75h15A2.25 2.25 0 0 1 21.75 12v.75m-8.69-6.44-2.12-2.12a1.5 1.5 0 0 0-1.061-.44H4.5A2.25 2.25 0 0 0 2.25 6v12a2.25 2.25 0 0 0 2.25 2.25h15A2.25 2.25 0 0 0 21.75 18V9a2.25 2.25 0 0 0-2.25-2.25h-5.379a1.5 1.5 0 0 1-1.06-.44Z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex items-center justify-end shrink-0 flex-wrap p-4 text-blue-gray-500"
            >
              <button
                class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gray-900 text-white shadow-md shadow-gray-900/10 hover:shadow-lg hover:shadow-gray-900/20 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none ml-auto"
                data-testid="submit-button"
                type="submit"
              >
                Test
              </button>
            </div>
          </form>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    data-floating-ui-portal=""
    id=":r9:"
  >
    <div
      style="position: fixed; overflow: auto; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 9999;"
    >
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="grid place-items-center fixed w-screen h-screen bg-black bg-opacity-60 backdrop-blur-sm"
        style="opacity: 0;"
      >
        <div
          aria-describedby=":r6:-description"
          aria-labelledby=":r6:-label"
          class="relative bg-white m-4 rounded-lg shadow-2xl text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed w-full md:w-2/3 lg:w-2/4 2xl:w-1/3 min-w-[80%] md:min-w-[66.666667%] lg:min-w-[50%] 2xl:min-w-[33.333333%] max-w-[80%] md:max-w-[66.666667%] lg:max-w-[50%] 2xl:max-w-[33.333333%] p-4"
          data-testid="library-manage-dialog"
          id=":r7:"
          role="dialog"
          style="opacity: 0; transform: translateY(-50px) translateZ(0);"
          tabindex="-1"
        >
          <form>
            <div
              class="items-center shrink-0 p-4 text-blue-gray-900 antialiased font-sans text-2xl font-semibold leading-snug relative m-0 block"
            >
              <h4
                class="block antialiased tracking-normal font-sans text-2xl font-semibold leading-snug text-blue-gray-900"
                data-testid="dialog-title"
              >
                Test title
              </h4>
              <p
                class="block antialiased font-sans text-base leading-relaxed mt-1 font-normal text-gray-600"
              >
                Test description
              </p>
              <button
                class="relative align-middle select-none font-sans font-medium text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-8 max-w-[32px] h-8 max-h-[32px] rounded-lg text-xs text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20 !absolute right-3.5 top-3.5"
                type="button"
              >
                <span
                  class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 stroke-2"
                    data-slot="icon"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 18 18 6M6 6l12 12"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="relative p-4 text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed space-y-4 pb-6"
            >
              <div>
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 mb-2 text-left font-medium"
                >
                  component.libraryManageDialog.name
                </p>
                <div
                  class="relative w-full min-w-[200px] h-11 !min-w-full"
                >
                  <input
                    class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent placeholder:opacity-0 focus:placeholder:opacity-100 text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-900 !border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900"
                    name="name"
                    placeholder="component.libraryManageDialog.namePlaceholder"
                  />
                  <label
                    class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-gray-500 peer-focus:text-gray-900 before:border-blue-gray-200 peer-focus:before:!border-gray-900 after:border-blue-gray-200 peer-focus:after:!border-gray-900 before:content-none after:content-none"
                  >
                     
                  </label>
                </div>
              </div>
              <div>
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 mb-2 text-left font-medium"
                >
                  component.libraryManageDialog.category
                </p>
                <div
                  class="relative w-full min-w-[200px] h-10"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal text-left outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all border text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 border-t-transparent"
                    name="category"
                    placeholder="1"
                    role="combobox"
                    type="button"
                  >
                    <span
                      class="absolute top-2/4 -translate-y-2/4 left-3 pt-0.5"
                      value="Movie"
                    >
                      component.libraryManageDialog.categoryOptions.movie
                    </span>
                    <div
                      class="grid place-items-center absolute top-2/4 right-2 pt-px w-5 h-5 text-blue-gray-400 rotate-0 -translate-y-2/4 transition-all"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                  <label
                    class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal transition-all -top-1.5 before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 before:rounded-tl-md before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 after:rounded-tr-md after:pointer-events-none after:transition-all peer-disabled:after:border-transparent text-[11px] peer-disabled:text-transparent before:border-t before:border-l after:border-t after:border-r leading-tight text-blue-gray-400 before:border-blue-gray-200 after:border-blue-gray-200"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-4"
              >
                <p
                  class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 text-left font-medium"
                >
                  component.libraryManageDialog.directory
                </p>
                <div
                  class="relative -mt-2 w-full"
                >
                  <div
                    class="relative w-full min-w-[200px] h-11 !min-w-full"
                  >
                    <input
                      class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent placeholder:opacity-0 focus:placeholder:opacity-100 text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-900 !border-t-blue-gray-200 placeholder:opacity-100 focus:!border-t-gray-900"
                      name="directory"
                      placeholder="component.libraryManageDialog.directoryPlaceholder"
                    />
                    <label
                      class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-gray-500 peer-focus:text-gray-900 before:border-blue-gray-200 peer-focus:before:!border-gray-900 after:border-blue-gray-200 peer-focus:after:!border-gray-900 before:content-none after:content-none"
                    >
                       
                    </label>
                  </div>
                  <div
                    class="absolute right-1.5 top-1.5"
                  >
                    <button
                      class="relative align-middle select-none font-sans font-medium text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-8 max-w-[32px] h-8 max-h-[32px] text-xs text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20 rounded"
                      disabled=""
                      type="button"
                    >
                      <span
                        class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform"
                      >
                        <svg
                          aria-hidden="true"
                          class="h-5 w-5"
                          data-slot="icon"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="1.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M2.25 12.75V12A2.25 2.25 0 0 1 4.5 9.75h15A2.25 2.25 0 0 1 21.75 12v.75m-8.69-6.44-2.12-2.12a1.5 1.5 0 0 0-1.061-.44H4.5A2.25 2.25 0 0 0 2.25 6v12a2.25 2.25 0 0 0 2.25 2.25h15A2.25 2.25 0 0 0 21.75 18V9a2.25 2.25 0 0 0-2.25-2.25h-5.379a1.5 1.5 0 0 1-1.06-.44Z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex items-center justify-end shrink-0 flex-wrap p-4 text-blue-gray-500"
            >
              <button
                class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gray-900 text-white shadow-md shadow-gray-900/10 hover:shadow-lg hover:shadow-gray-900/20 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none ml-auto"
                data-testid="submit-button"
                type="submit"
              >
                Test submit button text
              </button>
            </div>
          </form>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;
